card.go

```diff
@@ -1,49 +1,105 @@
 package goker
 
+import "fmt"
+
 type suit int8
-type rank int8
 
 const (
+	// Spade - Constant representing the suit ♠
 	Spade suit = iota
+	// Heart - Constant representing the suit ♥
 	Heart
+	// Diamond - Constant representing the suit ♦
 	Diamond
+	// Club - Constant representing the suit ♣
 	Club
 )
 
+type rank int8
+
 const (
+	// Two - Constant representing the rank 2
 	Two rank = iota + 2
+	// Three - Constant representing the rank 3
 	Three
+	// Four - Constant representing the rank 4
 	Four
+	// Five - Constant representing the rank 5
 	Five
+	// Six - Constant representing the rank 6
 	Six
+	// Seven - Constant representing the rank 7
 	Seven
+	// Eight - Constant representing the rank 8
 	Eight
+	// Nine - Constant representing the rank 9
 	Nine
+	// Ten - Constant representing the rank 10
 	Ten
+	// Jack - Constant representing the rank Jack
 	Jack
+	// Queen - Constant representing the rank Queen
 	Queen
+	// King - Constant representing the rank King
 	King
+	// Ace - Constant representing the rank Ace
 	Ace
 )
 
-type PlayingCard interface {
+// Card - Represents a playing card, with a suit and rank
+type Card interface {
 	Rank() rank
 	Suit() suit
+	String() string
 }
 
 type card struct {
 	rank rank
 	suit suit
 }
 
-func (c card) Rank() rank {
+func (c *card) Rank() rank {
 	return c.rank
 }
 
-func (c card) Suit() suit {
+func (c *card) Suit() suit {
 	return c.suit
 }
 
-func NewCard(r rank, s suit) card {
-	return card{r, s}
+func (c *card) String() string {
+	var suitStr string
+	switch c.Suit() {
```

Comment:
I see - so I should change the existing constants to structs with a dedicated field for their string representations. Then  `Card`'s `String()` method would just be
```
func (c Card) String() string {
  return c.rank.string + c.suit.string
}
```

Reply:
@aoeu I made some changes in bc5c5b0, but I'm not sure what you had in mind re: creating a suit struct. If I did that, I couldn't define the ranks and suits as constants, right?

Reply:
@sozorogami Yeah. For this, I've been trying to sketch out what that'd look like (suit or the ordinal values as structs vs. aliased types), which would omit the constants, and if that even makes sense as an approach: 

https://gist.github.com/aoeu/a48e10bda6911e90996bb63f10c9c73c

I don't know if that "code sketch" is the cleanest approach yet to representing cards, but there's a couple interesting qualities to it:

• There's a lack of constants, that are instead replaced with named variables, but that might be OK since all those variables are encapsulated (not exported) by the package.
• The only things that are actually exported are the Card type, and the interfaces to get "representations" of the underlying data - symbols that represent the card, English words that represent the card title, and a general string representation that dumps out that data.

I haven't been able to see how this approach would work out with hands, decks, and rankings yet, but I thought it might be still useful to share as just food-for-though. (Depending on what you have in mind for the library's usage, this approach might not be the best.)